docs/extending/creating_custom_dropdowns.md

```diff
@@ -5,82 +5,128 @@ Follow the steps below to learn how to integrate this component to fit it to you
+ `<select>` input shoud have `ibexa-input` class.
+
+Define `choices`:
+
+```twig
+{% set choices = [{
+    value: "DE",
+    label: "DE"
+}, {
+    value: "US",
+    label: "US"
+}, {
+    value: "NO",
+    label: "NO"
+}, {
+    value: "PL",
+    label: "PL"
+}] %}
+```
 
-```html
-    <select class="ibexa-custom-dropdown__select" hidden multiple></select>
+If you need a multilevel `choices`, use the following structure:
+
+```twig{% set choices = [{
+    value: "DE",
+    label: "DE"
+}, {
+    value: "US",
+    label: "US"
+}, {
+    value: "NO",
+    label: "NO"
+}, {
+    value: "PL",
+    label: "PL",
+    choices: [{
+        value: "PL_S",
+        label: "Silesian"
+    }, {
+        value: "PL_K",
+        label: "Kashubian"
+    }]
+}] %}
 ```
 
-![Drop-down multiple selection](img/dropdown_multiple_selection.jpg)
+To set `preferred_choices`, use the following:
 
-## Add attributes
+```twig
+{% set preferred_choices = [{
+    value: "NO",
+    label: "NO"
+}] %}
+```
 
-Next, look at the `data-value` attribute in the code (line 11 and 12) for duplicated options with the CSS class: `ibexa-custom-dropdown__item`.
-It stores a value of an option from a select input.
+For `value`, see the example:
 
-You can provide placeholder text for your custom drop-down. To do so:
+```twig
+{% set value = [{
+    value: "DE",
+    label: "DE"
+}] %}
+```
 
-- put a `data-value` attribute with no value `data-value=""`
-- add a `disabled` attribute to the item in the duplicated list of options to make it unclickable.
+You can configure the following attributes:
 
+Name|Values|Definition|
+|---|------|----------|
+|`source`| - |What is currently defined in the `<select>` input header.|
+|`choices`| - |Elements listed in the drop-down.|
+|`preferred_choices`|Elements listed at the top of the list with a separator.|
+|`value`|-|The currently selected element.It is an object with a key `value`. |
+|`multiple`| true</br>false|Boolean. To allow users to select multiple items, add this attribute to the same element.|
+|`translation_domain`|true</br>false|Used for translating choices and placeholder.|
+|`custom_form`|true</br>false|For custom form must be set to true.|
+|`class`| - |Additional classes for the element with `ibexa-dropdown` class.|
+`placeholder`|Displayed placeholder when no option is selected.|
+|`custom_init`|true</br>false|By default set to `false`. If set to `true`, requires to manually initialize drop-down in the JavaScript.|
 
-Example:  
- 
-```html
-<li data-value="" class="ibexa-custom-dropdown__item" disabled>Select an option</li>
-<li data-value="1" class="ibexa-custom-dropdown__item">Option 1</li>
-```
+![Dropd-own expanded state](img/dropdown_expanded_state.png)
 
 ## Initialize
 
-To initialize a custom drop-down, run the following JavaScript code:
+All drop-downs are searched and initialized automatically in `admin.dropdown.js`. However, if you want to extend it or make some modifications, run the following JavaScript code:
```

Comment:
I would add info that in order to extend or make modifications, there has to be added `custom_init` to twig element, otherwise it will be initialized two times (which is incorrect)